### PR TITLE
Add migrate mons module

### LINKED ISSUE #19: [module] migrate MONs

### PR DESCRIPTION
**Pull request by NotTheEvilOne**
This PR adds the module migrating Ceph mon daemons to Rook.

Closes #19

### CODE REVIEW

**Comment by boekhorstb1** [comment]
ok branch needs to be updated, rebase gives lots of conflicts

**Comment by boekhorstb1** [comment]
Please check: if I type nothing or random names in `config.yaml`, e.g. here:

```
migration_modules:
- bla
```

I get not warning and no outputs. Yet with `example` I get warning, for example if there is no ceph configuration in `.ceph`.

Will check later agan with more details

**Comment by NotTheEvilOne** [comment]
I added a separate issue (#56) for the described behavior as it is not part of the code changed with this PR.

File: src/rookify/modules/migrate_mons/main.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        migrated_mons += mon["name"]
+        self.machine.get_execution_state(
+            "MigrateMonsHandler"
+        ).migrated_mons = migrated_mons
+
+        self.logger.debug("Waiting for a quorum of 3 Ceph mon daemons")
+
+        while True:
+            result = self.ceph.mon_command("mon stat")
+            if result["num_mons"] >= 3:
```
Comment (brueggemann) [inline src/rookify/modules/migrate_mons/main.py:85]: I think the minimum number of running mons should be configurable via config file

**Comment by boekhorstb1** [comment]
Hi there!

ok the code worked after I deleted an existing configmap as describe in the issue here: https://github.com/SovereignCloudStack/rookify/issues/65

deletion is probably bad, I guess the check needs to be skipped if configmap allready exists @NotTheEvilOne

File: src/rookify/modules/migrate_mons/main.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                )
+            )
+
+        self.logger.debug(
+            "Waiting for disabled original Ceph mon daemon '{0}' to disconnect".format(
+                mon["name"]
+            )
+        )
+
+        while True:
```
Comment (brueggemann) [inline src/rookify/modules/migrate_mons/main.py:45]: We should check, if librados has a timeout for a mon connection, and automatically switches over to another mon, as we are probably shutting down the mon we are connected to.

**Review by boekhorstb1 — APPROVED** [review]

**Pull request merged**